client.go

```diff
@@ -1135,6 +1135,13 @@ func min(a, b int) int {
 	return a
 }
 
+func notExistOrErr(err error) error {
```

Comment:
That's fine.  I just followed the convention that was already in place in this section of the code.  Perhaps they could all be combined into one function?